Modules/Tracker/QuestieTracker.lua

```diff
@@ -989,13 +1011,244 @@ function QuestieTracker:Update()
+                        line.label:SetPoint("TOPLEFT", line, "TOPLEFT", objectiveMarginLeft, 0)
+
+                        if (criteriaString == "") then
+                            criteriaString = description
+                        end
+
+                        local objDesc = criteriaString:gsub("%.", "")
+
+                        -- [wasEarnedByMe] checks the "Parent" Achievement and if complete marks it complete.
+                        -- This allows listing Sub-Achievements embedded in an Achievement and treats them as
+                        -- Achievement objectives allowing us to list them as either complete or not complete.
+                        if wasEarnedByMe then
+                            line.label:SetText(Questie:Colorize(l10n("Achievement completed!"), "green"))
+                        elseif (not completed) then
+                            local lineEnding = tostring(quantityString)
+                            if lineEnding == "0" then
+                                line.label:SetText(QuestieLib:GetRGBForObjective({Collected=quantityProgress, Needed=quantityNeeded}) .. objDesc)
+                            else
+                                line.label:SetText(QuestieLib:GetRGBForObjective({Collected=quantityProgress, Needed=quantityNeeded}) .. objDesc .. ": " .. lineEnding)
+                            end
+
+                            if line.label:GetUnboundedStringWidth() > 290 then
+                                line.label:SetText(QuestieLib:GetRGBForObjective({Collected=quantityProgress, Needed=quantityNeeded}) .. objDesc .. ":\n" .. lineEnding)
+                                line.label:SetHeight(Questie.db.global.trackerFontSizeObjective * 3)
+                            end
+                        else
+                            line.label:SetText(QuestieLib:GetRGBForObjective({Collected=1, Needed=1}) .. objDesc)
+                        end
+
+                        local lineWidth = _QuestieTracker.baseFrame:GetWidth() - objectiveMarginLeft + 2 - trackerSpaceBuffer
+                        line.label:SetWidth(lineWidth)
+                        line:SetWidth(lineWidth)
+
+                        trackerLineWidth = math.max(trackerLineWidth, line.label:GetUnboundedStringWidth())
+
+                        line:SetVerticalPadding(1)
+                        line:Show()
+                        line.label:Show()
+                    end
+                end
+
+                if not line then
+                    line = LinePool.GetLastLine()
+                end
+
+                if line.label:GetHeight() > Questie.db.global.trackerFontSizeObjective + .01 then
+                    line:SetVerticalPadding(Questie.db.global.trackerFontSizeObjective + Questie.db.global.trackerQuestPadding)
+                else
+                    line:SetVerticalPadding(Questie.db.global.trackerQuestPadding)
+                end
+            end
+        end
+    end
```

Comment:
Could this be moved to a new function? The `:Update` function is already something like 300 lines long 🙈